pkg/api/api.go

```diff
@@ -182,49 +153,47 @@ type State struct {
 	// ActiveDevPC represents active DevPC controlling robot (0 … 90).
 	ActiveDevPC uint8 `json:"activedevpc"`
 
-	// Temperature1 represents temperature Motor 1 (0 … 99).
-	Temperature1 uint8 `json:"temperature_m1"`
-
-	// Temperature2 represents temperature Motor 2 (0 … 99).
-	Temperature2 uint8 `json:"temperature_m2"`
-
-	// Temperature3 represents temperature Motor 3 (0 … 99).
-	Temperature3 uint8 `json:"temperature_m3"`
-
-	// IsActive represents TRC is Active.
-	IsActive bool `json:"is_active"`
-
-	// CamStatus represents camera Status (0 … 10).
-	CamStatus uint8 `json:"cam_status"`
-
-	// LibSVNRevision represents TRC Library revision (0 … 99999).
-	LibSVNRevision uint32 `json:"libsvnrev"`
-
-	// CapacitorState represents capacitor state (No State/Empty/Full).
-	CapacitorState CapacitorState `json:"capacitorstate"`
-
 	// Kinect1State represents status of Kinect 1 (No State/No Ball/Ball).
 	Kinect1State KinectState `json:"kinect1_state"`
 
 	// Kinect2State represents status of Kinect 2 (No State/No Ball/Ball).
 	Kinect2State KinectState `json:"kinect2_state"`
-
-	// TODO: Figure out if IsActive corresponds to software being active or "robot"
-	// being active in e.g. penalty mode.
 }
 
 // Message specifies the type of the message.
 type MessageType string
 
 const (
-	MessageTypeGetState MessageType = "get_state"
-	MessageTypeSetState MessageType = "set_state"
-	MessageTypeCommand  MessageType = "command"
+	MessageTypeState     MessageType = "state"
+	MessageTypePing      MessageType = "ping"
+	MessageTypeHandshake MessageType = "handshake"
 )
 
+// Handshake represents the handshake message payload.
+type Handshake struct {
+	Version semver.Version `json:"version"`
+}
+
+// State represents the state of the TRC.
+type State struct {
+	Command Command                 `json:"command,omitempty"`
+	Turtles map[string]*TurtleState `json:"turtles,omitempty"`
+}
+
 // Message is the structure exchanged between TRC and SRRS.
 type Message struct {
 	Type      MessageType     `json:"type"`
 	MessageID ulid.ULID       `json:"message_id"`
+	ParentID  *ulid.ULID      `json:"parent_id,omitempty"`
 	Payload   json.RawMessage `json:"payload,omitempty"`
 }
+
+// NewMessage returns a new Message.
+func NewMessage(typ MessageType, pld json.RawMessage, parentID *ulid.ULID) *Message {
+	return &Message{
+		Type:      typ,
+		MessageID: ulid.MustNew(ulid.Now(), rand.Reader),
```

Comment:
rand.Reader is secure randomness or not secure randomness? In case it is not secure then you must ask yourself if it cannot be dangerous if an attacker might know what the next message id will be.

Reply:
It is cryptographically secure, but it's not required here.